Tests fail to build with boost >= 1.84.0

With boost 1.83.0 unit tests still build fine, but from 1.84.0 and later building fails. For example:

```
azmq> /build/source/test/cpp20/socket/main.cpp:32:18: error: 'BOOST_CURRENT_FUNCTION' was not declared in this scope; did you mean 'BOOST_CURRENT_LOCATION'?
azmq>    32 |     sb.bind(subj(BOOST_CURRENT_FUNCTION));
azmq>       |                  ^~~~~~~~~~~~~~~~~~~~~~
azmq>       |                  BOOST_CURRENT_LOCATION
```

GCC's suggestion is in this case wrong and instead one probably needs to `#include <boost/current_function.hpp>`. The file and macro is available [since boost 1.65.0](https://www.boost.org/doc/libs/1_65_0/libs/assert/doc/html/assert.html#current_function_macro_boost_current_function_hpp). Maybe it was transitively included in previous boost releases, but is not anymore so.

--- a/test/cpp20/socket/main.cpp
+++ b/test/cpp20/socket/main.cpp
@@ -14,6 +14,8 @@
 #include <boost/asio/awaitable.hpp>
 #include <boost/asio/use_awaitable.hpp>
 
+#include <boost/current_function.hpp>
+
 #include <coroutine>
 #include <array>
 #include <string>

--- a/test/socket/main.cpp
+++ b/test/socket/main.cpp
@@ -13,6 +13,7 @@
 #include <boost/algorithm/string/split.hpp>
 #include <boost/algorithm/string/classification.hpp>
 #include <boost/asio/buffer.hpp>
+#include <boost/current_function.hpp>
 
 #if BOOST_VERSION >= 107000
 #include <boost/asio/use_future.hpp>

--- a/test/socket_ops/main.cpp
+++ b/test/socket_ops/main.cpp
@@ -2,6 +2,7 @@
 #include <azmq/detail/socket_ops.hpp>
 
 #include <boost/asio/buffer.hpp>
+#include <boost/current_function.hpp>
 
 #include <array>
 #include <chrono>

